docs/zksync-error-howto.md

```diff
@@ -0,0 +1,377 @@
+# What is zksync-error?
+
+A generated crate describing all errors that are:
```

Comment:
It feels like this document tries to cover two things:
- What is `zksync-error`
- How to use it in `anvil-zksync` specifically.

IMHO, the first part should be a part of `zksync-error` repository, and here we would do something like "Please read this document first (link), and below we will cover how to use it for this codebase".

Otherwise, it will be a copypasta to be added to every repo and it will be a PITA to manage.

We can do it later if you prefer, however, I don't intend to block merging over this.

Reply:
Ideally, zksync-error must have a book published similarly to core/prover/compiler docs covering the architecture/usage/etc, but this can wait until we integrate it here.

Reply:
@dutterbutter if I recall correctly you suggested including project-agnostic documentation for zksync-error in this repository. If that was the case, could you remind the rationale?

Reply:
I think for the initial integration having it in this repo is fine given most of the early usage / work with it will be within anvil-zksync. I know for myself I will be referencing it and having it within the repo is certainly convenient. 

That being said, definitely once we've defined some standards / best practices it should be hosted and published separately. That will be done outside of this PR.

Reply:
1. I have started a mdbook here: https://github.com/matter-labs/zksync-error/tree/dev/docs
2. Let's continue working on this doc in one of the following PRs. Indeed we don't want every project to have a copy.